frontend/src/components/Reactions.tsx

```diff
@@ -0,0 +1,38 @@
+import {
+  FaThumbsUp,
+  FaThumbsDown,
+  FaDeaf,
+  FaLowVision,
+  FaHandPaper,
+  FaHandshake,
+  FaQuestionCircle,
+} from 'react-icons/fa';
+
+export const getReactions = (reactionsData: { [reactionId: string]: { [id: string]: any } }, userId: string | null) => {
+  const now = new Date();
+  const timeAgo = (dateString: string) => (now as any) - Date.parse(dateString);
+  const expiryTime = parseInt(process.env.REACT_APP_REACTION_EXPIRY_TIME as string, 10);
+  const getReaction = (reactionId: string, usersDates: { [id: string]: any }) => {
+    const freshReactions = Object.entries(usersDates).filter(([id, date]) => timeAgo(date) < expiryTime);
+    const isBlocked = userId && usersDates[userId] && timeAgo(usersDates[userId]) < expiryTime;
+    return { "count": freshReactions.length, "isBlocked": isBlocked };
+  };
+  return Object.fromEntries(Object.entries(reactionsData).map(([k, v]) => [k, getReaction(k, v)]));
+};
```

Comment:
a co byś powiedział na takie podejście. Front do backendu pozostaje bez zmian, ale za to backend pilnuje wygasania reakcji i wysyła do frontu dane w formacie
```
interface ReactionData {
  reaction: number;
  delta: number;
}
```
wysłanie reakcji to będzie rozesłanie z backendu `{reaction: ID, delta: 1}`, a potem `setTimeout`, który wysyła `{reaction: ID, delta: -1}`
Czy takie podejście nie byłoby prostsze?

Reply:
jeszcze `userId` do `ReactionData` aby móc zablokować i odblokować reakcję jeśli to nasz idik jest w danych

Reply:
no byłoby dużo prostsze - jedyny problem o którym myślałem to sytuacja w której w jakiś magiczny sposób ten sam user wyśle 2 razy daną reakcję w ciągu tych 10s (zaloguje się na innym urządzeniu/z innej karty i to źle obsłużę lub będzie jakiś bug na froncie) - tak jak jest to teraz napisane gwarantuje że aktywność jednego usera może przełożyć się na co najwyżej +1/-1

Reply:
ale dzięki -to podejście rzeczywiście by było dużo prostsze xd